lib/discourse_antivirus/clamav.rb

```diff
@@ -41,38 +48,33 @@ def update_versions
     end
 
     def accepting_connections?
-      sockets = clamav_services_pool.all_tcp_sockets
-
-      if sockets.empty?
-        update_status(true)
-        return false
-      end
+      unavailable = clamav_services_pool.all_offline?
 
-      available = sockets.reduce(true) { |memo, socket| memo && target_online?(socket) }
+      PluginStore.set(PLUGIN_NAME, UNAVAILABLE, unavailable)
 
-      available.tap do |status|
-        unavailable = !status
-        update_status(unavailable)
-      end
+      !unavailable
     end
 
     def scan_upload(upload)
-      begin
-        file = get_uploaded_file(upload)
+      file = get_uploaded_file(upload)
 
-        return error_response(DOWNLOAD_FAILED) if file.nil?
+      return error_response(DOWNLOAD_FAILED) if file.nil?
 
-        scan_file(file)
-      rescue OpenURI::HTTPError
-        error_response(DOWNLOAD_FAILED)
-      rescue StandardError => e
-        Rails.logger.error("Could not scan upload #{upload.id}. Error: #{e.message}")
-        error_response(e.message)
-      end
+      scan_file(file)
+    rescue OpenURI::HTTPError
+      error_response(DOWNLOAD_FAILED)
+    rescue StandardError => e
+      Rails.logger.error("Could not scan upload #{upload.id}. Error: #{e.message}")
+      error_response(e.message)
     end
 
     def scan_file(file)
-      scan_response = with_session { |socket| stream_file(socket, file) }
+      online_service = clamav_services_pool.find_online_service
+
+      # We open one connection to check if the service is online and another
+      # to scan the file. Code belows check if one of them failed with `&` and `unless`:
```

Comment:
I find the second part of the comment here unnecessary "Code below..." because it just explains the code.